pkg/cmd/rollbackcopy/fileioutils.go

```diff
@@ -0,0 +1,152 @@
+package rollbackcopy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"k8s.io/klog"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+func dirExists(dirname string) bool {
+	info, err := os.Stat(dirname)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return info.IsDir()
+}
+
+func safeDirRename(src, dest string, srcMayNotExist bool) error {
+	if !dirExists(src) {
+		if !srcMayNotExist {
+			return fmt.Errorf("SafeDirRename: src dir %s does not exist", src)
+		} else {
+			return nil
+		}
+	}
+	destExists := false
+	destToBeRemoved := dest + ".to_be_removed"
+	if dirExists(dest) {
+		destExists = true
+		if dirExists(destToBeRemoved) {
+			os.RemoveAll(destToBeRemoved)
+		}
+		if err := os.Rename(dest, destToBeRemoved); err != nil {
+			return fmt.Errorf("safeDirRename: got error renaming %s %s. Err: %w", dest, destToBeRemoved, err)
+		}
+	}
+
+	if err := os.Rename(src, dest); err != nil {
+		if destExists {
+			os.Rename(destToBeRemoved, dest)
+		}
+		return fmt.Errorf("safeDirRename: got error renaming %s %s. Err: %w", src, dest, err)
+		return err
+	}
+	klog.Info("Successfully moved ", src, " to ", dest)
+	os.RemoveAll(destToBeRemoved) // ignore the error
+	return nil
+}
+
+// copyDir copies a whole directory recursively
+
+func findTheLatestRevision(dir, podname string) (string, error) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return "", err
+	}
+
+	var modTime time.Time
+	var latest string
+	found := false
+	for _, f := range files {
+		if f.IsDir() && strings.HasPrefix(f.Name(), podname) {
+			if f.ModTime().After(modTime) {
+				modTime = f.ModTime()
+				latest = f.Name()
+				found = true
+			}
+		}
+	}
+	if !found {
+		return "", fmt.Errorf("Not found any resources for pod %s", podname)
+	}
+	return filepath.Join(dir, latest), nil
+}
+
+type backupVersion struct {
+	ClusterVersion string `json:"ClusterVersion"`
+	TimeStamp      string `json:"TimeStamp"`
+}
+
+func versionChanged(dir1, dir2 string) bool {
+	dir1Version, err := getVersion(dir1)
+	if err != nil {
+		klog.Errorf("getVersion failed on %s", dir1)
+		return false
+	}
+	dir2Version, err := getVersion(dir2)
+	if err != nil {
+		klog.Errorf("getVersion failed on %s", dir2)
```

Comment:
As above, I think treating a failure to determine the version as "the version is unchanged" is probably incorrect or at least prone to causing subtle bugs for callers